Application won't start correctly in dev mode for Postgres

<!--
- Please follow the issue template below for bug reports.
- If you have a support request rather than a bug, please use [Stack Overflow](http://stackoverflow.com/questions/tagged/jhipster) with the JHipster tag.
- For bug reports it is mandatory to run the command `jhipster info` in your project's root folder, and paste the result here.
- Tickets opened without any of these pieces of information will be **closed** without any explanation.
-->

##### **Overview of the issue**

<!-- Explain the bug, if an error is being thrown a stack trace helps -->
Application won't start correctly in dev mode for Postgres because liquibase is not loading the fake data correctly.
Somehow relationship column value is being inserted as string value and hence having this error.
```
2019-05-04 00:06:25.729 ERROR 10688 --- [datatype-task-1] liquibase.changelog.ChangeSet            : Change Set config/liquibase/changelog/20190503173054_added_entity_Entity2.xml::20190503173054-1-data::jhipster failed.  Error: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO public.entity2(id, name, entity1_id) VALUES(1, 'mobile Fish', '8') was aborted: ERROR: column "entity1_id" is of type bigint but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 65  Call getNextException to see other errors in the batch.
2019-05-04 00:06:25.736 ERROR 10688 --- [datatype-task-1] i.g.j.c.liquibase.AsyncSpringLiquibase   : Liquibase could not start correctly, your database is NOT ready: Migration failed for change set config/liquibase/changelog/20190503173054_added_entity_Entity2.xml::20190503173054-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO public.entity2(id, name, entity1_id) VALUES(1, 'mobile Fish', '8') was aborted: ERROR: column "entity1_id" is of type bigint but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 65  Call getNextException to see other errors in the batch.

liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20190503173054_added_entity_Entity2.xml::20190503173054-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO public.entity2(id, name, entity1_id) VALUES(1, 'mobile Fish', '8') was aborted: ERROR: column "entity1_id" is of type bigint but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 65  Call getNextException to see other errors in the batch.
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:637)
	at liquibase.changelog.visitor.UpdateVisitor.visit(UpdateVisitor.java:53)
	at liquibase.changelog.ChangeLogIterator.run(ChangeLogIterator.java:83)
	at liquibase.Liquibase.update(Liquibase.java:202)
	at liquibase.Liquibase.update(Liquibase.java:179)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:353)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:305)
	at io.github.jhipster.config.liquibase.AsyncSpringLiquibase.initDb(AsyncSpringLiquibase.java:100)
	at io.github.jhipster.config.liquibase.AsyncSpringLiquibase.lambda$afterPropertiesSet$0(AsyncSpringLiquibase.java:80)
	at io.github.jhipster.async.ExceptionHandlingAsyncTaskExecutor.lambda$createWrappedRunnable$1(ExceptionHandlingAsyncTaskExecutor.java:68)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: liquibase.exception.DatabaseException: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO public.entity2(id, name, entity1_id) VALUES(1, 'mobile Fish', '8') was aborted: ERROR: column "entity1_id" is of type bigint but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 65  Call getNextException to see other errors in the batch.
	at liquibase.statement.ExecutablePreparedStatementBase.execute(ExecutablePreparedStatementBase.java:86)
	at liquibase.executor.jvm.JdbcExecutor.execute(JdbcExecutor.java:115)
	at liquibase.database.AbstractJdbcDatabase.execute(AbstractJdbcDatabase.java:1229)
	at liquibase.database.AbstractJdbcDatabase.executeStatements(AbstractJdbcDatabase.java:1211)
	at liquibase.changelog.ChangeSet.execute(ChangeSet.java:600)
	... 12 common frames omitted
Caused by: java.sql.BatchUpdateException: Batch entry 0 INSERT INTO public.entity2(id, name, entity1_id) VALUES(1, 'mobile Fish', '8') was aborted: ERROR: column "entity1_id" is of type bigint but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 65  Call getNextException to see other errors in the batch.
	at org.postgresql.jdbc.BatchResultHandler.handleError(BatchResultHandler.java:148)
	at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:824)
	at org.postgresql.jdbc.PgPreparedStatement.executeBatch(PgPreparedStatement.java:1538)
	at com.zaxxer.hikari.pool.ProxyStatement.executeBatch(ProxyStatement.java:128)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.executeBatch(HikariProxyPreparedStatement.java)
	at liquibase.statement.BatchDmlExecutablePreparedStatement.executePreparedStatement(BatchDmlExecutablePreparedStatement.java:65)
	at liquibase.statement.ExecutablePreparedStatementBase.execute(ExecutablePreparedStatementBase.java:84)
	... 16 common frames omitted
Caused by: org.postgresql.util.PSQLException: ERROR: column "entity1_id" is of type bigint but expression is of type character varying
  Hint: You will need to rewrite or cast the expression.
  Position: 65
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2440)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2183)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:308)
	at org.postgresql.jdbc.PgStatement.executeBatch(PgStatement.java:821)
	... 21 common frames omitted
```

##### **Motivation for or Use Case**

<!-- Explain why this is a bug for you -->
Application should start normally in dev mode.

##### **Reproduce the error**

<!-- For bug reports, an unambiguous set of steps to reproduce the error -->

1. Create a new app with postgres as both prod and dev DB

1. Use following jdl
```
// JDL definition for application 'datatype' generated with command 'jhipster export-jdl'

entity Entity1 {
  name String
}
entity Entity2 {
  name String
}
relationship ManyToOne {
  Entity2{entity1(name) required} to Entity1
}

dto Entity1, Entity2 with mapstruct
service Entity1, Entity2 with serviceImpl

```

3. `./mvnw`

##### **Related issues**

<!-- Has a similar issue been reported before? Please search both closed & open issues -->

##### **Suggest a Fix**

-
<!-- If you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit) -->

##### **JHipster Version(s)**

```
datatype@0.0.0 /Users/sierra/github/temp/datatype
└── generator-jhipster@6.0.0 

```


##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**


<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.mycompany.myapp"
    },
    "jhipsterVersion": "6.0.0",
    "applicationType": "monolith",
    "baseName": "datatype",
    "packageName": "com.mycompany.myapp",
    "packageFolder": "com/mycompany/myapp",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "no",
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "postgresql",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": false
  }
}
</pre>
</details>


##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity Entity1 {
  name String
}
entity Entity2 {
  name String
}
relationship ManyToOne {
  Entity2{entity1(name) required} to Entity1
}

dto Entity1, Entity2 with mapstruct
service Entity1, Entity2 with serviceImpl

</pre>
</details>


##### **Environment and Tools**

java version "1.8.0_172"
Java(TM) SE Runtime Environment (build 1.8.0_172-b11)
Java HotSpot(TM) 64-Bit Server VM (build 25.172-b11, mixed mode)

git version 2.20.1 (Apple Git-117)

node: v10.14.2

npm: 6.7.0

yeoman: 2.0.6

Docker version 18.09.1, build 4c52b90

docker-compose version 1.23.2, build 1110ad01

INFO! Congratulations, JHipster execution is complete!
-   [x] Checking this box is mandatory (this is just to show you read everything)

<!-- Love JHipster? Please consider supporting our collective:
👉  https://opencollective.com/generator-jhipster/donate -->

--- a/generators/entity-server/templates/src/main/resources/config/liquibase/changelog/added_entity.xml.ejs
+++ b/generators/entity-server/templates/src/main/resources/config/liquibase/changelog/added_entity.xml.ejs
@@ -197,6 +197,16 @@
             <column name="<%=fields[idx].fieldNameAsDatabaseColumn%>_content_type" type="string"/>
                 <%_ } _%>
             <%_ } _%>
+            <%_ for (idx in relationships) {
+                    let loadColumnType = 'numeric';
+                    if (relationships[idx].relationshipValidate === true && relationships[idx].relationshipRequired) {
+                        let baseColumnName = getColumnName(relationships[idx].relationshipName) + '_id';
+                        if (relationships[idx].otherEntityNameCapitalized === 'User' && authenticationType === 'oauth2') {
+                            loadColumnType = 'varchar(100)';
+                        } _%>
+            <column name="<%=baseColumnName%>" type="<%=loadColumnType%>"/>
+                <%_ } _%>
+            <%_  } _%>
         </loadData>
     </changeSet>
 

